client-app/app/components/env-tab.js

```diff
@@ -1,27 +1,61 @@
 import Component from "@ember/component";
 import { computed } from "@ember/object";
 import { buildHashString } from "client-app/lib/utilities";
+import Preload from "client-app/lib/preload";
 
 export default Component.extend({
   current: 1,
 
   didUpdateAttrs() {
     this.set("current", 1);
+    this.set("expanded", null);
   },
 
   isEnvArray: computed("message.env", function() {
     return Array.isArray(this.get("message.env"));
   }),
 
-  html: computed("isEnvArray", "current", function() {
+  html: computed("isEnvArray", "current", "expanded.[]", function() {
     if (!this.get("isEnvArray")) {
       return buildHashString(this.get("message.env"));
     } else {
-      const currentEnv = this.get("message.env")[this.get("current") - 1];
-      return buildHashString(currentEnv);
+      const currentEnv = Em.$.extend(
+        {},
+        this.get("message.env")[this.get("current") - 1]
+      );
+      const expandableKeys = Preload.get("env_expandable_keys");
+      expandableKeys.forEach(key => {
+        if (currentEnv.hasOwnProperty(key) && !Array.isArray(currentEnv[key])) {
+          const list = [currentEnv[key]];
+          this.get("message.env").forEach(env => {
+            if (env[key] && list.indexOf(env[key]) === -1) {
+              list.push(env[key]);
+            }
+          });
+          currentEnv[key] = list.length > 1 ? list : list[0];
+        }
+      });
+      return buildHashString(currentEnv, false, this.get("expanded") || []);
     }
   }),
 
+  click(e) {
+    const $elem = Em.$(e.target);
+    const dataKey = $elem.data("key");
```

Comment:
I usually do `$elem.attr("data-key")` for this reason https://stackoverflow.com/a/8708345

data() is not bad, but I think attr("data-xxx") has a more expected behaviour.